File: src/main/java/model/equipment/weapon/Hammer.java
package model.equipment.weapon;

import model.support.allEnums.DmgType;
import model.support.allEnums.WeaponType;

public class Hammer extends Weapon {
    public Hammer(double weight, int quality) {
        super(WeaponType.HAMMER, weight, quality);
        isOneHanded = true;
        setDamage();
    }

    private void setDamage() {
        damage.put(DmgType.BLUNT, quality * 0.10);
        damage.put(DmgType.PIERCING, quality * 0.90);
    }
}

File: src/main/java/model/equipment/weapon/Hammer.kt
package model.equipment.weapon

import model.support.enums.Skill
import model.support.enums.Skill.*
import model.support.enums.Specification.*
import model.support.enums.WeaponType.HAMMER

class Hammer(weight: Double, quality: Int) : OneHandedWeapon(HAMMER, HAMMER_SKILL, ONE_HANDED_WEAPON_SKILL, weight, quality, true) {
    init {
        damage = mapOf(
                BLUNT to quality * 0.10,
                CUT to 0.0,
                SLASH to 0.0,
                PIERCING to quality * 0.90,
                CRUSH to 0.0)
    }
}
